### DISCUSSION TITLE
Imap, connection refuse

### DISCUSSION THREAD

**Discussion by leoborlot** [roundcube/roundcubemail-docker#232] [General]
Hi gyus, ts po

I can´t connect with an imap 993.
If I use anothe imap client, this works... But ievery time return Timeout, and ths logsay:

errors: <1c3f9a3f> IMAP Error: Login failed for teste@emailfacil.com.br against srv01.emailfacil.com.br from 172.19.0.2 (X-Real-IP: 179.217.19.80,X-Forwarded-For: 179.217.19.80). Could not connect to ssl://srv01.emailfacil.com.br:993: Connection timed out in /var/www/html/program/lib/Roundcube/rcube_imap.php on line 211 (POST /?_task=login&_action=login)
172.19.0.2 - - [03/Nov/2023:02:28:48 +0000] "POST /?_task=login HTTP/1.1" 401 3113 "https://mail.emailfacil.com.br/?_task=login" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36

Theres any way to fix this:?

**Comment by leoborlot** [comment]
Update.

Now we have connection, but, don access. 

If  I config as:
      - ROUNDCUBEMAIL_DEFAULT_HOST=ssl://srv01.emailfacil.com.br
      - ROUNDCUBEMAIL_DEFAULT_PORT=993

recive in Roundcube:

```
errors: <285d4a11> IMAP Error: Login failed for teste@emailfacil.com.br against srv01.emailfacil.com.br from 172.19.0.2 (X-Real-IP: 179.217.19.80,X-Forwarded-For: 179.217.19.80). Empty startup greeting (srv01.emailfacil.com.br:993) in /var/www/html/program/lib/Roundcube/rcube_imap.php on line 211 (POST /?_task=login&_action=login)
172.19.0.2 - - [03/Nov/2023:16:02:21 +0000] "POST /?_task=login HTTP/1.1" 401 3115 "https://mail.emailfacil.com.br/?_task=login" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36"
```

and recive on imap dovecot:
```
Nov  3 12:57:38 srv01 dovecot: imap(teste@emailfacil.com.br)<5558><bcJKkEEJ1MysEwAE>: Disconnected: Logged out in=90 out=977 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
Nov  3 12:57:51 srv01 dovecot: imap-login: Disconnected: Connection closed: SSL_accept() failed: error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca: SSL alert number 48 (no auth attempts in 0 secs): user=<>, rip=172.19.0.4, lip=172.19.0.9, TLS handshaking: SSL_accept() failed: error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca: SSL alert number 48, session=<mn0JkUEJmresEwAE>
```

If I config as:
      - ROUNDCUBEMAIL_DEFAULT_HOST=tls://srv01.emailfacil.com.br
      - ROUNDCUBEMAIL_DEFAULT_PORT=993
      
i recive Timeout response in Roudcube.

```
errors: <285d4a11> IMAP Error: Login failed for teste@emailfacil.com.br against srv01.emailfacil.com.br from 172.19.0.2 (X-Real-IP: 179.217.19.80,X-Forwarded-For: 179.217.19.80). Empty startup greeting (srv01.emailfacil.com.br:993) in /var/www/html/program/lib/Roundcube/rcube_imap.php on line 211 (POST /?_task=login&_action=login)
172.19.0.2 - - [03/Nov/2023:16:02:21 +0000] "POST /?_task=login HTTP/1.1" 401 3115 "https://mail.emailfacil.com.br/?_task=login" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36"
```
     
and recive on imap dovecot:
     
    ` ov  3 13:00:36 srv01 dovecot: imap-login: Disconnected: Connection closed (no auth attempts in 60 secs): user=<>, rip=172.19.0.4, lip=172.19.0.9, TLS handshaking: Connection closed, session=<52XemkEJwtSsEwAE> `

if I config as:
      - ROUNDCUBEMAIL_DEFAULT_HOST=srv01.emailfacil.com.br
      - ROUNDCUBEMAIL_DEFAULT_PORT=993

I recive in Roundcube:

```
errors: <285d4a11> IMAP Error: Login failed for teste@emailfacil.com.br against srv01.emailfacil.com.br from 172.19.0.2 (X-Real-IP: 179.217.19.80,X-Forwarded-For: 179.217.19.80). Empty startup greeting (srv01.emailfacil.com.br:993) in /var/www/html/program/lib/Roundcube/rcube_imap.php on line 211 (POST /?_task=login&_action=login)
172.19.0.2 - - [03/Nov/2023:16:02:21 +0000] "POST /?_task=login HTTP/1.1" 401 3115 "https://mail.emailfacil.com.br/?_task=login" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36"
```

I recive in imap dovecot: 

```
Nov  3 12:57:51 srv01 dovecot: imap-login: Disconnected: Connection closed: SSL_accept() failed: error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca: SSL alert number 48 (no auth attempts in 0 secs): user=<>, rip=172.19.0.4, lip=172.19.0.9, TLS handshaking: SSL_accept() failed: error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca: SSL alert number 48, session=<mn0JkUEJmresEwAE>
```

**Reply by Galloway10002** [reply]
I am getting the same error.

If I shell into the roundcube container I can connect and log in to the email address manually from the command line.  This works:

```root@24d614275bfb:/var/www/html# openssl s_client -connect host.containers.internal:993 -crlf```

On the other hand, if I try to log in using the web interface, here is the error roundcube gives:

`[roundcubemail] | errors: <2b178f61> IMAP Error: Login failed for questionc@questionc.net against host.containers.internal from 10.89.1.2. Could not connect to ssl://host.containers.internal:993: Unknown reason in /var/www/html/program/lib/Roundcube/rcube_imap.php on line 197 (POST /?_task=login&_action=login)
`

Here is the error on the mail server side:
```

2026-08-01T02:20:43.203005+02:00 mail dovecot: imap-login: Disconnected: Connection closed (disconnected before auth was ready, waited 0 secs): user=<>, rip=144.91.106.29, lip=10.89.0.3, TLS, session=<AICVRvFXcsqQW2od>
2026-08-01T02:20:47.894411+02:00 mail postfix/anvil[659955]: statistics: max connection rate 1/60s for (submission:46.151.182.202) at Aug  1 02:12:58
2026-08-01T02:20:47.894435+02:00 mail postfix/anvil[659955]: statistics: max connection count 1 for (submission:46.151.182.202) at Aug  1 02:12:58
2026-08-01T02:20:47.894441+02:00 mail postfix/anvil[659955]: statistics: max cache size 20 at Aug  1 02:14:18
2026-08-01T02:21:53.576751+02:00 mail postfix/submission/smtpd[660760]: connect from unknown[46.151.182.202]
2026-08-01T02:21:53.679097+02:00 mail postfix/submission/smtpd[660760]: disconnect from unknown[46.151.182.202] ehlo=1 auth=0/1 rset=0/1 quit=1 commands=2/4
```

It  connects to the database and the web interface works, it just errors when I try to log in.

From `compose.yaml`:
```
      - ROUNDCUBEMAIL_DEFAULT_HOST=ssl://host.containers.internal
      - ROUNDCUBEMAIL_DEFAULT_PORT=993
      - ROUNDCUBEMAIL_SMTP_SERVER=ssl://host.containers.internal
      - ROUNDCUBEMAIL_SMTP_PORT=465
```

**Comment by kelvin-sim** [comment]
hi, just wondering if you managed to resolve this?
I'm having exactly the same problem.

**Reply by pabzm** [reply]
Port 993 is usually imaps, which requires SSL (not plain, not TLS). The first error contains "alert unknown ca", which hints to an untrusted certificate or incomplete certificate chain somewhere.

Please try to connect to the containerized dovecot-service with another client to find out if the services works at all.

**Reply by kelvin-sim** [reply]
I can connect from thunderbird (android and windows) client no issues.  I've tested it with another android client too and it works fine.
I only get this error when I try to connect it from roundcubemail

**Reply by pabzm** [reply]
As mentioned, reading the originally posted log lines I would suspect a self-signed (or otherwise publicly untrusted) TLS certificate.

What the problem in your case is I cannot guess without more details.

**Reply by kelvin-sim** [reply]
my question is more for @leoborlot to find out if he managed to resolved his.  

I will raise a separate discussion for mine if I need more detailed help.  FYI, it's not a self-signed cert.
